lufopyramids.sh

```diff
@@ -16,81 +23,205 @@
-# done
+# When using ECW files, translate to GeoTiff manually:
+# https://hub.docker.com/r/ginetto/gdal
+# docker run --rm -it --name gdalecw -v /mnt/lufo/lufo2009/ams:/home/datafolder ginetto/gdal:2.4.4_ECW gdal_translate /home/datafolder/Amsterdam_groot.ecw /home/datafolder/lufo2009.tif
 #
-# cp -a *.aux /mnt/lufo2020
-# cp -a *.tfw /mnt/lufo2020
 
+### Set local vars
 SOURCEDIR=$1
 DESTDIR_ROOT=$2
 DESTDIR=$2/pyramid
 YEAR=$3
 TYPE=$4
 
-# Create a list of images in the source directory
+### Reset all tifs to use 3 bands:
+for i in $SOURCEDIR/*.tif; do
+    [ -f "$i" ] || break
+    gdal_translate $i $DESTDIR_ROOT/$(basename $i) -a_srs "EPSG:28992" -b 1 -b 2 -b 3
+done
+
+# ### Create a list of images in the source directory
+gdalbuildvrt -overwrite -resolution highest -r cubic -hidenodata -srcnodata "255 255 255" $DESTDIR_ROOT/overview.vrt $DESTDIR_ROOT/*.tif
 
-if [ -f /tmp/list.txt ] ; then
-    rm -r /tmp/list.txt
+### Clean up
+if [ -d $DESTDIR ] ; then
+   rm -rf $DESTDIR
 fi
 
-ls $SOURCEDIR/*.tif > /tmp/list.txt
+### Create pyramid base dir and retile to create pyramid with 5 levels in fixed tile sizes
+mkdir -p $DESTDIR/0
+gdal_retile.py -v -r cubic -resume -ps 8192 8192 -co TILED=YES -co COMPRESS=JPEG -co PHOTOMETRIC=YCBCR -s_srs "EPSG:28992" -targetDir $DESTDIR $DESTDIR_ROOT/overview.vrt
+# gdal_retile.py -v -r cubic -levels 5 -resume -ps 8192 8192 -co TILED=YES -co COMPRESS=JPEG -co PHOTOMETRIC=YCBCR -s_srs "EPSG:28992" -targetDir $DESTDIR $DESTDIR_ROOT/overview.vrt
 
-#if [ -d $DESTDIR ] ; then
-#    rm -rf $DESTDIR
-#fi
+### Put original resolution images in their own subdirectory
+mv $DESTDIR/*.tif $DESTDIR/0
 
-mkdir -p $DESTDIR/0
-gdal_retile.py -v -r cubic -levels 5 -ps 8192 8192 -co TILED=YES -co COMPRESS=JPEG -co PHOTOMETRIC=YCBCR -s_srs "EPSG:28992" -targetDir $DESTDIR --optfile /tmp/list.txt
+### Set Levels and loop over them
+### This exists to ensure public WMS usage is much quicker as the TIF files are in the blob storage
+LEVELS=$(find $DESTDIR/* -type d -exec basename {} \; | sort)
+for i in $LEVELS; do
+
+### Create overviews to show when zoomed out of original resolution
+    for j in $DESTDIR/$i/*.tif; do
+        gdaladdo -r cubic --config COMPRESS_OVERVIEW JPEG --config PHOTOMETRIC_OVERVIEW YCBCR --config INTERLEAVE_OVERVIEW PIXEL $j 2 4 8 16 32 64 128
+    done
 
-# Put original resolution images in their own subdirectory
+### Create tile indices for Mapserver WMS usage
+    gdaltindex $DESTDIR/imagery-$YEAR-$i.shp $DESTDIR/$i/*.tif
+    ogrinfo -dialect SQLITE -sql "UPDATE 'imagery-$YEAR-$i' SET location = '/vsicurl/https://files.data.amsterdam.nl/$TYPE/$YEAR/pyramid' || SUBSTR(location,length('$DESTDIR/'),length(location))" $DESTDIR/imagery-$YEAR-$i.shp
+    ogrinfo -sql "CREATE SPATIAL INDEX ON imagery-$YEAR-$i" $DESTDIR/imagery-$YEAR-$i.shp
+
+done
+
+### Create tile index for Mapserver Tiling only
+gdaltindex $DESTDIR_ROOT/imagery.shp $DESTDIR/0/*.tif
```

Comment:
Duplicate of https://github.com/Amsterdam/mapserver/blob/steph/lufo_updates/lufopyramids.sh#L106. Make sure to create an index file using `ogrinfo -sql "CREATE SPATIAL INDEX ON..."(sse L108) and maybe add proper location according to L107).